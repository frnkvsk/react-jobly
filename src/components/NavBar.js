import React, { useContext } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { Link } from 'react-router-dom';
// import { LoginContext } from '../context/LoginContext';
// import { useLogin } from '../hooks/useLogin';
import { AuthContext } from '../context/AuthContext';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    alignItems: 'center',
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
    // fontSize: '20px',
    color: '#ffffff',
    textDecoration: 'none',
  },
}));

export default function NavBar() {
  const classes = useStyles();
  // const {loginStatus, setLoginStatus} = useContext(LoginContext);
  // const {setTokenStorage} = useLogin();
  const auth = useContext(AuthContext);
  console.log('NavBar auth=',auth)
  const handleClick = () => {
    // setLoginStatus(false);    
    // setTokenStorage("");
    auth.setAuthState({
      token: "",
      userInfo: {}
    });
  }

  return (
    <div className={classes.root}>
      {/* {console.log('NavBar loginStatus',loginStatus)} */}
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h5" className={classes.title} component={Link} to={"/"} >
            Jobly
          </Typography>
          {auth.authState.token !== "" ? <>
            <Button color="inherit" component={Link} to={"/companies"} >Companies</Button>
            <Button color="inherit" component={Link} to={"/jobs"} >Jobs</Button>
            <Button color="inherit" component={Link} to={"/profile"} >Profile</Button>
            <Button onClick={handleClick} color="inherit" component={Link} to={"/login"} >Log out</Button>
            </> :
            <Button color="inherit" component={Link} to={"/login"} >login / signup</Button>
          }
        </Toolbar>
      </AppBar>
    </div>
  );
}

