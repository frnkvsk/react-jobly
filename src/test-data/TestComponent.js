// import React, { useContext } from 'react';
// import { AuthContext } from '../context/AuthContext';

// const TestComponent = () => {
//   const auth = useContext(AuthContext);
//   const setAuth2 = async () =>
//     await auth.setAuthState({
//       token: 'testtoken',
//       userInfo: {
//       username: 'testuser',
//       first_name: 'testfirst',
//       last_name: 'testlast',
//       email: 'test@test.com',
//       photo_url: null
//     }});
//   (async () => await setAuth2())();
//   console.log('-------auth.authState',auth.authState)
//   return (
//     <div>
//       <div>{auth.authState.userInfo.username}</div>
//       <div>{auth.authState.userInfo.first_name}</div>
//       <div>{auth.authState.userInfo.last_name}</div>
//       <div>{auth.authState.userInfo.email}</div>
//       <div>{auth.authState.userInfo.photo_url}</div>
//     </div>
//   )
// }
// export default TestComponent;

