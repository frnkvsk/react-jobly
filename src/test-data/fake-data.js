const testData = {
  jobs: [
    {
    company_handle: "test-company-a",
    equity: 0.115,
    id: 1,
    salary: 11111,
    state: null,
    title: "Editor, magazine features",
    },
      {
      company_handle: "test-company-b",
      equity: 0.22,
      id: 2,
      salary: 22222,
      state: null,
      title: "Coder",
    },
      {
      company_handle: "test-company-c",
      equity: 0.33,
      id: 3,
      salary: 33333,
      state: null,
      title: "Front-end Dev",
    },
  ],
  companies: [
    {
      description: "Test description 1",
      handle: "test-company-1",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 1",
    },
    {
      description: "Test description 2",
      handle: "test-company-2",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 2",
    },
    {
      description: "Test description 3",
      handle: "test-company-3",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 3",
    },
  ],
}

export { testData }