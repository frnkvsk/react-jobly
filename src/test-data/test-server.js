import { rest } from 'msw';
import { setupServer } from 'msw/node';

let server;

const setServer = async (url, status, message='') => {
  if(server) {
    server.close();
    server.resetHandlers();
  }
  server = await setupServer(
    rest.get(url, (req, res, ctx) => {
      return res(
        ctx.status(status),
        ctx.json(message)
      )
    })
  );
}
  

beforeAll(() => server ? server.listen() : null);
afterAll(() => server ? server.close() : null);
afterEach(() => server ? server.resetHandlers() : null);

export { setServer, server, rest }
