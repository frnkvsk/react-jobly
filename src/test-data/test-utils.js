// test-utils.js
import React from 'react';
import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';
import { SearchProvider } from '../context/SearchContext';
import { AuthProvider } from '../context/AuthContext';
import { PageCountProvider } from '../context/PageCountContext';

const AllTheProviders = (page) => {
  return render(
    <BrowserRouter>
      <AuthProvider>
        <SearchProvider>
          <PageCountProvider>
            {page}
          </PageCountProvider>
        </SearchProvider>
      </AuthProvider> 
    </BrowserRouter>
    
  );
}

// re-export everything
export * from '@testing-library/react';

// override render method
export { AllTheProviders as render }
