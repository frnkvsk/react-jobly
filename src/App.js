import React, { useState } from 'react';
import { BrowserRouter, Route, Switch, useParams } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import NavBar from './components/NavBar';
import Home from './pages/Home';
import Company from './pages/Company';
import Companies from './pages/Companies';
import Jobs from './pages/Jobs';
import Profile from './pages/Profile';
import Login from './pages/Login';
import Page404 from './pages/Page404';
import { SearchContext } from './context/SearchContext';
import { LoginContext } from './context/LoginContext';
import { useLogin } from './hooks/useLogin';

const useStyles = makeStyles((theme) => ({
  root: {
    // height: '100vh',
    backgroundColor: '#f5f5f5',
  },
  
}));
function CompanyHandle() {
  let {handle} = useParams();
  console.log('App CompanyHandle handle=',handle)
  return <Company handle={handle} />
}

function App() {
  const classes = useStyles();
  const [token] = useLogin();
  console.log('App token',token)
  const [search, setSearch] = useState("");
  
  const [loginStatus, setLoginStatus] = useState(token ? true : false);

  return (
      <BrowserRouter>
        <LoginContext.Provider value={{loginStatus, setLoginStatus}}>
          <NavBar loggedin={true} />
          <main className={classes.root}>
            <Switch>
              <Route exact path="/">
                <Home />          
              </Route>
              <Route exact path="/companies/:handle"> 
                <CompanyHandle />
              </Route>
              <Route exact path="/companies">
                <SearchContext.Provider value={{search, setSearch}}>
                  <Companies />
                </SearchContext.Provider>              
              </Route>
              <Route exact path="/jobs">
                <SearchContext.Provider value={{search, setSearch}}>
                  <Jobs />
                </SearchContext.Provider>    
              </Route>
              <Route path="/profile">
                <Profile />
              </Route>
              <Route exact path="/login">
                <Login login={"signup"} />
              </Route>
              <Route>
                <Page404 />
              </Route>
            </Switch>
          </main>
        </LoginContext.Provider>
        
      </BrowserRouter>
  );
}

export default App;
