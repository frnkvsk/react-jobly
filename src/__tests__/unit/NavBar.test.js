import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, fireEvent } from '../../test-data/test-utils';
import NavBar from '../../components/NavBar';
import App from '../../App';

// smoke and appearance tests
test('renders NavBar ', async() => {
  render(<NavBar />)
  
  expect(screen.getByText(/login \/ signup/i)).toBeInTheDocument(); 

  // fireEvent does't cause error
  fireEvent.click(screen.getByRole('button', {name: /login \/ signup/i}));

  expect(screen.getByRole('link', {name: /jobly/i})).toBeInTheDocument();

  // fireEvent does't cause error
  fireEvent.click(screen.getByRole('link', {name: /jobly/i}));
  
});

// test NavBar through App 
test('link and button changes App appearance ', async() => {
  render(<App />)
  
  expect(screen.getByText(/login \/ signup/i)).toBeInTheDocument();
  
  fireEvent.click(screen.getByRole('button', {name: /login \/ signup/i}));

  expect(screen.getByText(/username/i)).toBeInTheDocument();
  
  fireEvent.click(screen.getByRole('link', {name: /jobly/i}));

  expect(screen.getByText(/All the jobs in one, convenient place./i)).toBeInTheDocument();
});


// snapshot test
test('matches snapshot', () => {
  const { asFragment } = render(<NavBar />);
  expect(asFragment()).toMatchSnapshot();
});