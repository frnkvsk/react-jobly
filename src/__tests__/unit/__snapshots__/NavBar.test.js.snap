// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-29"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <a
          class="MuiTypography-root makeStyles-title-31 MuiTypography-h5"
          href="/"
        >
          Jobly
        </a>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          href="/login"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            login / signup
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
